### ISSUE TITLE
keras.callbacks.ModelCheckpoint modified but still saving weights for every epoch

### ISSUE THREAD

**Issue by wiamadaya** [matterport/Mask_RCNN#1604]
i have modified the following in model.py, but during training, weights are still saved for every epoch, i have limited disk space.

keras.callbacks.ModelCheckpoint(self.checkpoint_path, verbose=0, save_best_only=True, save_weights_only=True, mode='auto', period=5)

**Comment by cpoptic** [comment]
Can confirm this is an issue.  I have several models saved in `.h5` format via the ModelCheckpoint callback, but when I go to load the model using keras.models.load_model(), I get the dreaded `Cannot create group in read only mode` error, which indicates that the .h5 files did NOT contain the model architecture. loss, etc.

The problem is that keras models offer no obvious way to check whether they contain the model weights + architecture until you actually try loading the `.h5` file.
Frustrating, as without the rest of the model architecture, the saved model cannot be loaded properly.

This is a HUGE bug in the ModelCheckpoint callback.

**Comment by Shakesbeer333** [comment]
Modifying the model.py file gives you the desired result: 
```
        # Callbacks
        modelcheckpoint_dir = '/content/gdrive/My Drive/weights/'
        
        if not os.path.exists(modelcheckpoint_dir):
            os.mkdir(modelcheckpoint_dir)
        
        callbacks = [
            keras.callbacks.TensorBoard(log_dir=self.log_dir,
                                        histogram_freq=0, write_graph=True, write_images=True),
            keras.callbacks.ModelCheckpoint(modelcheckpoint_dir + 'best_model.h5',
                                            verbose=1, save_weights_only=True, save_best_only=True),
            keras.callbacks.EarlyStopping(monitor='val_loss', mode='min', verbose=1, patience=10)
        ]
```

Note: If you're not interested in early stopping, just comment it out
